pkg/test/integration/common/framework.go

```diff
@@ -615,11 +692,46 @@ func (c *IntegrationTestFramework) SetupBeforeSuite() {
 	ginkgo.By("Checking for the clusters if provided are available")
 	gomega.Expect(c.initalizeClusters()).To(gomega.BeNil())
 
-	// preparing resources
-	// if control cluster is not the seed, then applyCrds from the mcm repo by cloning
-	// if no image tags specified, then also clone the mcm repo as the the mcm process needs to be started
+	//setting up MCM either locally or by deploying after checking conditions
+
+	/* flow dealing with MCM deployment
+
+		- Check if control cluster is a seed cluster
+	    - If YES
+	        - Check if MC Image or MCM Image are set in the Makefile
+	            - If YES (that is, one of them or both of them are passed)
+	                - Prepare MCM Deployment in the Seed cluster by updating the images of the containers
+	            - If NO (that is both of them are not passed)
+	                - Clone the MCM repo
+	                - Scale down MCM running in seed to 0
+	                - Run MCM locally
+	    - If NO
+	        - Then, most likely this is a shoot whose original MCM is also running. Scale it down (this is challenging)
+	        - Apply the CRDs (Clone MCM repo before this as CRDs present in MCM repo)
+	        - Check if MC Image or MCM Image are set in the Makefile
+	            - If YES (that is, one of them or both of them are passed)
+	                - Prepare MCM Deployment in the control cluster by creating (/kubernetes/out-of-tree/deployment.yaml)
+	            - If NO (that is both of them are not passed)
+	                - Run MCM locally
```

Comment:
You can remove these reference comments once everything is working satisfactorily.